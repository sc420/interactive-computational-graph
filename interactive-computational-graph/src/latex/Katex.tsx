import katex from "katex";
import { useMemo, type FunctionComponent } from "react";

interface KatexProps {
  latex: string;
}

const Katex: FunctionComponent<KatexProps> = ({ latex }) => {
  const options = useMemo((): katex.KatexOptions => {
    return {
      output: "mathml",
      throwOnError: false,
    };
  }, []);

  return (
    <span
      dangerouslySetInnerHTML={{ __html: katex.renderToString(latex, options) }}
    ></span>
  );
};

export default Katex;
