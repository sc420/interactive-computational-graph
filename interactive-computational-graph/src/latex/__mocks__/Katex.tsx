import { type FunctionComponent } from "react";

interface KatexProps {
  latex: string;
}

const KatexMock: FunctionComponent<KatexProps> = ({ latex }) => {
  return <span>{latex}</span>;
};

export default KatexMock;
