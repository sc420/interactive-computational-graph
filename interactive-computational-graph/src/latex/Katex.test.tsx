import { render } from "@testing-library/react";
import Katex from "./Katex";

test("should render constant", () => {
  const { container } = render(<Katex latex="3.14" />);

  expect(container).toMatchSnapshot();
});

test("should render displaystyle", () => {
  const { container } = render(
    <Katex latex="\displaystyle =\frac{\partial{a}}{\partial{b}} \cdot \frac{\partial{c}}{\partial{d}}" />,
  );

  expect(container).toMatchSnapshot();
});
