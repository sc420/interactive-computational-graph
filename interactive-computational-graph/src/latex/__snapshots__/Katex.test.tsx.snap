// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render constant 1`] = `
<div>
  <span>
    <span
      class="katex"
    >
      <math
        xmlns="http://www.w3.org/1998/Math/MathML"
      >
        <semantics>
          <mrow>
            <mn>
              3.14
            </mn>
          </mrow>
          <annotation
            encoding="application/x-tex"
          >
            3.14
          </annotation>
        </semantics>
      </math>
    </span>
  </span>
</div>
`;

exports[`should render displaystyle 1`] = `
<div>
  <span>
    <span
      class="katex"
    >
      <math
        xmlns="http://www.w3.org/1998/Math/MathML"
      >
        <semantics>
          <mrow>
            <mstyle
              displaystyle="true"
              scriptlevel="0"
            >
              <mo>
                =
              </mo>
              <mfrac>
                <mrow>
                  <mi
                    mathvariant="normal"
                  >
                    ∂
                  </mi>
                  <mi>
                    a
                  </mi>
                </mrow>
                <mrow>
                  <mi
                    mathvariant="normal"
                  >
                    ∂
                  </mi>
                  <mi>
                    b
                  </mi>
                </mrow>
              </mfrac>
              <mo>
                ⋅
              </mo>
              <mfrac>
                <mrow>
                  <mi
                    mathvariant="normal"
                  >
                    ∂
                  </mi>
                  <mi>
                    c
                  </mi>
                </mrow>
                <mrow>
                  <mi
                    mathvariant="normal"
                  >
                    ∂
                  </mi>
                  <mi>
                    d
                  </mi>
                </mrow>
              </mfrac>
            </mstyle>
          </mrow>
          <annotation
            encoding="application/x-tex"
          >
            \\displaystyle =\\frac{\\partial{a}}{\\partial{b}} \\cdot \\frac{\\partial{c}}{\\partial{d}}
          </annotation>
        </semantics>
      </math>
    </span>
  </span>
</div>
`;
